includes/class-amp-theme-support.php

```diff
@@ -470,6 +471,36 @@ public static function add_amp_component_scripts() {
 		echo self::COMPONENT_SCRIPTS_PLACEHOLDER; // phpcs:ignore WordPress.Security.EscapeOutput, WordPress.XSS.EscapeOutput
 	}
 
+	/**
+	 * Overrides the output of 'Video' widgets for YouTube and Vimeo.
+	 *
+	 * WP_Widget_Media_Video::render() has a different output for YouTube and Vimeo.
+	 * It calls wp_video_shortcode(), instead of wp_oembed_get() like the rest of the videos.
+	 * So this callback overrides the output of wp_video_shortcode().
+	 * It produces 'youtube' or 'vimeo' shortcodes,
+	 * which this plugin's embed handlers render as <amp-youtube> or <amp-vimeo>.
+	 * This sometimes returns an empty string: ''.
+	 * If the value is anything other than '', wp_video_shortcode() exits and returns the value.
+	 *
+	 * @param string $html Markup to possibly override.
+	 * @param array  $attr The shortcode attributes.
+	 * @return string $markup The markup to output.
+	 */
+	public static function video_override( $html, $attr ) {
```

Comment:
In other words, add this method to both the `AMP_Vimeo_Embed_Handler` and `AMP_YouTube_Embed_Handler` classes, but in each one only account for the Vimeo or YouTube URLs respectively.

Reply:
Thanks, @westonruter. This [commit](https://github.com/Automattic/amp-wp/pull/921/commits/7c1f8fe9f80fc7c948433596e60f602639843093) from above applies your suggestion.